### PR TITLE
BT-31137: Added optional filters to additional fields in table2gui

### PR DESCRIPTION
**Pull request by M-Barz**
Link to Mantis-Report:
https://mantis.ilias.de/view.php?id=31137

### CODE REVIEW

File: Services/Table/classes/class.ilTable2GUI.php [hunk 1]
```diff
@@ -292,10 +292,19 @@ public function determineSelectedColumns()
                 if ($new_column) {
                     $set = true;
                 }
-                if ($c["default"]) {
+                if (isset($c["default"])) {
```
Comment (smeyer-ilias) [inline Services/Table/classes/class.ilTable2GUI.php:295]: This is a non backward compatible change:
$c[default] = false
=> the if evaluates to false.

File: Services/Table/classes/class.ilTable2GUI.php [hunk 2]
```diff
@@ -281,7 +281,7 @@ public function determineSelectedColumns()
 
             $new_column = ($sel_fields[$k] === null);
 
-            if ($_POST["tblfsh" . $this->getId()]) {
+            if (isset($_POST["tblfsh" . $this->getId()])) {
```
Comment (smeyer-ilias) [inline Services/Table/classes/class.ilTable2GUI.php:284]: Not backward compatible.

File: Services/Table/classes/class.ilTable2GUI.php [hunk 3]
```diff
@@ -292,10 +292,19 @@ public function determineSelectedColumns()
...
                     $set = true;
                 }
-                if ($c["default"]) {
+                if (isset($c["default"])) {
                     $this->selected_column[$k] = true;
                 }
             }
+
+            // Optional filters
+            if (isset($_POST["tblff" . $this->getId()])) {
```
Comment (smeyer-ilias) [inline Services/Table/classes/class.ilTable2GUI.php:301]: Please check whether isset is correct here.

**Comment by mjansenDatabay** [comment]
@M-Barz / @smeyer-ilias I think this changed the current behaviour: If a column was set to `'default' => false`, it will **now** be initially displayed. I observed this in ILIAS 6, where this code is applied since : https://github.com/ILIAS-eLearning/ILIAS/commit/e0b137feefc5d22bb29cff14ddd3ded071c95d74

**Comment by smeyer-ilias** [comment]
@mjansenDatabay : yes, see my comments half an hour ago ;-)

**Pull request merged**